lib/tapioca/dsl/compilers/smart_properties.rb

```diff
@@ -86,8 +86,9 @@ class << self
           def gather_constants
             all_modules.select do |c|
               name_of(c) &&
-                c != ::SmartProperties::Validations::Ancestor &&
-                c < ::SmartProperties && ::SmartProperties::ClassMethods === c
+                ::SmartProperties > c &&
```

Comment:
Checking this first will disqualifier more constants earlier, evaluating fewer operands in this `&&` chain.

It also lines things up nicely :)